woocommerce/woocommerce-android
# Observe last update my store stats

Closes: #9443

### Why
As part of the Reliability project, we are working on improving the app's performance and reducing the perceived time it takes to load. One of the ideas we are working on is introducing a timestamp as the web does and preventing redundant requests to the API. This PR is part of the timestamp introduction in the My Store screen.

This how the web looks like:
<img width="594" alt="Screenshot 2023-06-21 at 11 25 43" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/133b162b-f9f9-4344-bdb1-577f07cbbb43">

### Description
This PR adds the last update value to the My Store screen. Now every time the last update value changes, the UI will be refreshed.

### Testing instructions
TC1
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Check that the last update value is shown
4. Press on a different granularity (ex. This week)
5. Wait until the screen finish loading
6. Check that the last update value is shown
7. Change back to the first selection (Today)
8. Check that no request is made to the API and that data is presented from the cache
9. Check that the last update value is shown with the same value as in Step 3
10. Change to the second selection (This week)
11. Check that no request is made to the API and that data is presented from the cache
12. Check that the last update value is shown with the same value as Step 6

TC2
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Check that the last update value is shown
4. Press on a different granularity (ex. This week)
5. Wait until the screen finish loading
6. Check that the last update value is shown
7. Change back to the first selection (Today)
8. Check that the last update value is shown as in Step 3
9. Check that no request is made to the API and that data is presented from the cache
10. Pull-To-Refresh and check that a call is made to the API and the data is refreshed
11. Check that last update value is updated to current time

### Images/gif
<img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/b9a87215-881b-40da-a868-a095f9d4613d" />

https://github.com/woocommerce/woocommerce-android/assets/18119390/a85e45f3-a179-4808-9cd4-0330584a8e90
